src/js/builtins/Module.ts

```diff
@@ -34,45 +38,70 @@ export function overridableRequire(this: CommonJSModuleRecord, id: string) {
     $evaluateCommonJSModule(existing);
     return existing.exports;
   }
-
-  if (id.endsWith(".node")) {
+  const filename = id;
+  if (filename.endsWith(".node")) {
     return $internalRequire(id);
   }
-
   // To handle import/export cycles, we need to create a module object and put
   // it into the map before we import it.
   const mod = $createCommonJSModule(id, {}, false, this);
   $requireMap.$set(id, mod);
 
+  // @ts-ignore - TS doesn't know each function is isolated.
+  const Module = require("node:module");
+  const { _extensions } = Module;
+  if (_extensions && _extensions.$structureChanged) {
```

Comment:
this is going to be very slow. we should not use a custom dynamic getter.

Reply:
> this is going to be very slow. we should not use a custom dynamic getter.

Is this more expensive than say any other JS related thing like `propertyIsEnumerable`cor `String#slice`? If it's comparable it may be okay.

Reply:
Measuring with `require('@babel/core')` showed no noticeable change in variance in the debug build (though I didn't stress test it).

Reply:
@paperdave Is there a scenario you're interested in?

Reply:
Moving to c++ helper.